CMakeLists.txt

```diff
@@ -62,13 +62,13 @@ option(INSTALL_LIB "Install the library" OFF)
 tr_auto_option(ENABLE_DEPRECATED "Allow deprecated API use of upstream packages, e.g. GTK" AUTO)
 tr_auto_option(RUN_CLANG_TIDY "Run clang-tidy on the code" OFF)
 tr_auto_option(USE_SYSTEM_EVENT2 "Use system event2 library" AUTO)
-tr_auto_option(USE_SYSTEM_DEFLATE "Use system deflate library" AUTO)
-tr_auto_option(USE_SYSTEM_DHT "Use system dht library" AUTO)
-tr_auto_option(USE_SYSTEM_MINIUPNPC "Use system miniupnpc library" AUTO)
-tr_auto_option(USE_SYSTEM_NATPMP "Use system natpmp library" AUTO)
-tr_auto_option(USE_SYSTEM_UTP "Use system utp library" AUTO)
-tr_auto_option(USE_SYSTEM_B64 "Use system b64 library" AUTO)
-tr_auto_option(USE_SYSTEM_PSL "Use system psl library" AUTO)
+tr_auto_option(USE_SYSTEM_DEFLATE "Use system deflate library" OFF)
```

Comment:
This is starting to become a pattern (there're 2 such instances already), and I don't like it. The whole point of using `tr_auto_option()` is to default to `AUTO`, admittedly I made a mistake designing it as one shouldn't even have to pass this 3rd argument. Once you make the default something that is not `AUTO`, I'm having a hard time imagining someone willing to set it to `AUTO` when configuring (if you're smart enough to pass the option, you're smart enough to know if you have the need system package installed). That is to say, if you want to default to `OFF`, use plain `option()` instead.

What's more, since previous behavior was to prefer (not default to, as you say, which is what `ON` is for) system libraries, CI has to be adjusted to preserve those (or similar) test configurations, so that they don't break in the future. This is important to package maintainers. What's happening right now is CI installs dev packages that are then not being used.

Last but not least, I don't understand why libevent was made an exception. We are bundling it just as all the other dependencies, and defaulting to the bundled version should work on all the platforms, unless you know something I don't.

Reply:
> This is starting to become a pattern (there're 2 such instances already)

I am not a fan of this either but FWIW the two places where we currently default to `OFF` it wasn't done arbitrarily:

- Turning off clang-tidy by default to work around an upstream issue discussesd in #4821

- Turning off web rebuilding because defaulting to `AUTO` there requires network access for `npm install` and availability of a modern version of Node.js.

I don't see any similar rationale for these other changes though? :thinking:

Reply:
Definitely different than those you've listed, Charles. It was just a suggestion for the reasons cited above. I thought reducing the different build variables might even the playing field for bugs or issues reported. That's cool. I'l close.